src/solidity-frontend/solidity_grammar.cpp

```diff
@@ -65,21 +91,19 @@ TypeNameT get_type_name_t(const nlohmann::json &type_name)
   if(type_name.contains("typeString"))
   {
     // for AST node that contains ["typeName"]["typeDescriptions"]
-    if(type_name["typeString"] == "uint8" || type_name["typeString"] == "bool")
+    std::string typeString = type_name["typeString"].get<std::string>();
+
+    if(uint_string_to_type_map.count(typeString) || typeString == "bool")
     {
       // For state var declaration,
       return ElementaryTypeName;
     }
-    else if(
-      type_name["typeString"].get<std::string>().find("int_const") !=
-      std::string::npos)
+    else if(typeString.find("int_const") != std::string::npos)
```

Comment:
Sorry, I'm not familiar with the Solidity language nor the translated JSON very much. Is it possible to define own type names there? Could I for instance define a `float` (if it exists) to be called `my_int_const`? If yes, then this condition using `.find()` will (probably wrongly) recognize my "typedef'ed" `float` as an `int_const`. Is that a problem that can occur with Solidity?

Reply:
I believe solidity doesn't support type aliases, so these type strings are from a predefined set of keywords.